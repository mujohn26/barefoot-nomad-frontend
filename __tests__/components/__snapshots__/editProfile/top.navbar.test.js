import React from 'react';
import { shallow } from 'enzyme';
import {
	TopNavBar,
	mapStateToProps,
} from '../../../../src/components/profile/top.nav.bar';
import { props } from '../../../../__mockData__/notification.mock.data';

describe('Top navigation bar component', () => {
	it('should handle mapStateToProps successfully', () => {
		const initState = {
			NotificationReducer: {
				Notifications: [],
				markNotificationAsRead: [],
				limitNumber: [10],
				notificationPaneHeight: [500],
				viewButton: [true],
				buttonTitle: 'more',
			},
		};
		expect(mapStateToProps(initState)).toBeTruthy();
	});

	it('should render the top navigation bar  component successfully', () => {
		const wrapper = shallow(<TopNavBar {...props} />);
		expect(wrapper).toMatchSnapshot();
	});
	it('should render notifications', () => {
		const wrapper = shallow(<TopNavBar {...props} />);
		const findNotifications = jest.spyOn(wrapper.instance(), 'count');
		wrapper.instance().count(props.notifications);
		expect(findNotifications).toHaveBeenCalled();
	});
	it('should render notifications which is equal to zero', () => {
		const wrapper = shallow(<TopNavBar {...props} />);
		const findNotifications = jest.spyOn(wrapper.instance(), 'count');
		wrapper.instance().count(props.notification);
		expect(findNotifications).toHaveBeenCalled();
	});
	it('should render notifications greater than nine', () => {
		const wrapper = shallow(<TopNavBar {...props} />);
		const findNotifications = jest.spyOn(wrapper.instance(), 'count');
		wrapper.instance().count(props.notificationGreaterThanNine);
		expect(findNotifications).toHaveBeenCalled();
	});
});
