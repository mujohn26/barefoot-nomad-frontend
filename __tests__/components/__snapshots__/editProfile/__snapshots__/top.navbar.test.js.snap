// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top navigation bar component should render the top navigation bar  component successfully 1`] = `
<Fragment>
  <Styled(MuiBox)
    display="flex"
    justifyContent="flex-end"
    width={0.3}
  >
    <Styled(MuiBox)
      pr={1}
      style={
        Object {
          "marginTop": "10px",
        }
      }
    >
      <img
        height="24px"
        src="https://res.cloudinary.com/dby88h516/image/upload/v1580893608/email_1_jupvlq.svg"
        width="100%"
      />
    </Styled(MuiBox)>
    <Styled(MuiBox)
      pl={1}
      style={
        Object {
          "marginTop": "10px",
        }
      }
    >
      <PopupState
        popupId="demo-popup-popover"
        style={
          Object {
            "width": "50%",
          }
        }
        variant="popover"
      >
        <Component />
      </PopupState>
    </Styled(MuiBox)>
  </Styled(MuiBox)>
</Fragment>
`;
